### PR TITLE
Remove `utils/documentation_tests.txt`

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Let's remove `utils/documentation_tests.txt` and just keep `not_doctested.txt`

### CODE REVIEW

File: utils/tests_fetcher.py [hunk 1]
```diff
@@ -396,26 +424,19 @@ def get_doctest_files(diff_with_last_commit: bool = False) -> List[str]:
...
         slow_documentation_tests = set(fp.read().strip().split("\n"))
-
-    # So far we don't have 100% coverage for doctest. This line will be removed once we achieve 100%.
-    test_files_to_run = [x for x in test_files_to_run if x not in not_doctested and x not in slow_documentation_tests]
-
-    # The file `utils/not_doctested.txt` doesn't contain all files that are not doc-tested, so we need more filters.
-    # 1. only include files in `src` or `docs/source/en/`
-    test_files_to_run = [x for x in test_files_to_run if x.startswith(("src/", "docs/source/en/"))]
-    # 2. not include init files
-    test_files_to_run = [x for x in test_files_to_run if not x.endswith(("__init__.py",))]
```
Comment (ydshieh) [inline utils/tests_fetcher.py:413]: same comment

File: utils/tests_fetcher.py [hunk 2]
```diff
@@ -367,6 +367,34 @@ def get_diff_for_doctesting(repo: Repo, base_commit: str, commits: List[str]) ->
     return code_diff
 
 
+def get_all_doctest_files() -> List[str]:
+    """
+    Return the complete list of python and Markdown files on which we run doctest.
+
+    At this moment, we restrict the files to in `src/` or `docs/source/en/` that are not in `utils/not_doctested.txt`.
```
Comment (ArthurZucker) [inline utils/tests_fetcher.py:374]:
```suggestion
    At this moment, we restrict this to only take files from `src/` or `docs/source/en/` that are not in `utils/not_doctested.txt`.
```

File: utils/tests_fetcher.py [hunk 3]
```diff
@@ -367,6 +367,34 @@ def get_diff_for_doctesting(repo: Repo, base_commit: str, commits: List[str]) ->
...
+
+    Returns:
+        `List[str]`: The complete list of Python and Markdown files on which we run doctest.
+    """
+    py_files = [str(x.relative_to(PATH_TO_REPO)) for x in PATH_TO_REPO.glob("**/*.py")]
+    md_files = [str(x.relative_to(PATH_TO_REPO)) for x in PATH_TO_REPO.glob("**/*.md")]
+    test_files_to_run = py_files + md_files
+
+    # only include files in `src` or `docs/source/en/`
+    test_files_to_run = [x for x in test_files_to_run if x.startswith(("src/", "docs/source/en/"))]
```
Comment (ydshieh) [inline utils/tests_fetcher.py:384]: We have `docs/source/es/quicktour.md` tested so far with `documentation_tests` on daily CI. But the logic becomes a bit complex if to deal with language other than `en`. Let me know your preference.

File: utils/tests_fetcher.py [hunk 4]
```diff
@@ -367,6 +367,34 @@ def get_diff_for_doctesting(repo: Repo, base_commit: str, commits: List[str]) ->
...
 
 
+def get_all_doctest_files() -> List[str]:
+    """
+    Return the complete list of python and Markdown files on which we run doctest.
+
+    At this moment, we restrict the files to in `src/` or `docs/source/en/` that are not in `utils/not_doctested.txt`.
+
+    Returns:
+        `List[str]`: The complete list of Python and Markdown files on which we run doctest.
```
Comment (ArthurZucker) [inline utils/tests_fetcher.py:377]:
```suggestion
        `List[str]`: The filtered list of Python and Markdown files on which we run doctest.
```

File: .github/workflows/doctests.yml [hunk 5]
```diff
@@ -43,9 +43,13 @@ jobs:
...
         run: pip freeze
 
+      - name: Get doctest files
+        run: |
+          $(python3 -c 'from utils.tests_fetcher import get_all_doctest_files; to_test = get_all_doctest_files(); to_test = " ".join(to_test); fp = open("doc_tests.txt", "w"); fp.write(to_test); fp.close()')
+
       - name: Run doctests
         run: |
-          python3 -m pytest -v --make-reports doc_tests_gpu --doctest-modules $(cat utils/documentation_tests.txt) -sv --doctest-continue-on-failure --doctest-glob="*.md"
+          python3 -m pytest -v --make-reports doc_tests_gpu --doctest-modules $(cat doc_tests.txt) -sv --doctest-continue-on-failure --doctest-glob="*.md"
```
Comment (ydshieh) [inline .github/workflows/doctests.yml:52]: use the new `get_all_doctest_files` defined below

File: utils/tests_fetcher.py [hunk 6]
```diff
@@ -367,6 +367,34 @@ def get_diff_for_doctesting(repo: Repo, base_commit: str, commits: List[str]) ->
     return code_diff
 
 
+def get_all_doctest_files() -> List[str]:
```
Comment (ydshieh) [inline utils/tests_fetcher.py:370]: for daily CI to use

**Review by ArthurZucker — APPROVED** [review]
Very cool. Hope we won't have surprises in the ci daily

File: utils/tests_fetcher.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (ydshieh) [reply utils/tests_fetcher.py:370]: I will keep the current name here. It is indeed have a filtration, but it's fine as they are the complete files that we run doctest on daily CI. And in CircleCI, we have one more filtration to not include some know slow files.

File: utils/tests_fetcher.py (hunk 4 above)
Comment (ydshieh) [reply utils/tests_fetcher.py:377]: If we add `filtered`, it's also confusing that what `get_doctest_files` means - that is used for CircleCI and have one more filtration

**Pull request merged**